packages/notification-center-vue/src/lib/notification-center-component/NotificationCenterComponent.vue

```diff
@@ -1,53 +1,143 @@
-  notificationClicked?: NotificationCenterComponentProps['notificationClicked'];
-  unseenCountChanged?: NotificationCenterComponentProps['unseenCountChanged'];
-  actionClicked?: NotificationCenterComponentProps['actionClicked'];
-  tabClicked?: NotificationCenterComponentProps['tabClicked'];
-}>();
+export interface INotificationCenterComponentProps {
+  backendUrl?: NotificationCenterContentComponentProps['backendUrl'];
+  socketUrl?: NotificationCenterContentComponentProps['socketUrl'];
+  subscriberId?: NotificationCenterContentComponentProps['subscriberId'];
+  applicationIdentifier: NotificationCenterContentComponentProps['applicationIdentifier'];
+  subscriberHash?: NotificationCenterContentComponentProps['subscriberHash'];
+  stores?: NotificationCenterContentComponentProps['stores'];
+  tabs?: NotificationCenterContentComponentProps['tabs'];
+  showUserPreferences?: NotificationCenterContentComponentProps['showUserPreferences'];
+  popover?: {
+    offset?: number;
+    position?: FloatingPosition;
+  };
+  theme?: NotificationCenterContentComponentProps['theme'];
+  styles?: NotificationCenterContentComponentProps['styles'];
+  colorScheme?: NotificationCenterContentComponentProps['colorScheme'];
+  i18n?: NotificationCenterContentComponentProps['i18n'];
+  sessionLoaded?: NotificationCenterContentComponentProps['sessionLoaded'];
+  notificationClicked?: NotificationCenterContentComponentProps['notificationClicked'];
+  unseenCountChanged?: NotificationCenterContentComponentProps['unseenCountChanged'];
+  actionClicked?: NotificationCenterContentComponentProps['actionClicked'];
+  tabClicked?: NotificationCenterContentComponentProps['tabClicked'];
+}
+
+customElements.define('notification-center-content-component', NotificationCenterContentWebComponent);
+
+const props = withDefaults(defineProps<INotificationCenterComponentProps>(), {
+  colorScheme: 'dark',
+});
+const slots = useSlots();
+const popper = ref();
+const unseenCount = ref<number | undefined>(undefined);
+const hasSlot = ref(!!slots.default);
+
+const computedStyles = computed(() => {
+  const { popoverArrowClass, popoverDropdownClass, bellButtonClass, gradientDotClass, bellColors } = calculateStyles({
+    theme: props.theme,
+    styles: props.styles,
+    colorScheme: props.colorScheme,
+  });
+
+  return {
+    popoverArrowClass,
+    popoverDropdownClass,
+    bellButtonClass,
+    gradientDotClass,
+    bellColors,
+  };
+});
+
+const updateArrowStyles = (arrowClass: string) => {
+  // add the popover arrow class to the arrow element, there is no better way to do this
+  const arrowClasses = ['v-popper__arrow-outer', 'v-popper__arrow-inner'];
+  const arrowsContainer = popper.value.$refs.popper.$_arrowNode as HTMLDivElement;
+  arrowsContainer.childNodes.forEach((node, idx) => {
+    (node as HTMLDivElement).className = '';
+    (node as HTMLDivElement).classList.add(arrowClasses[idx], arrowClass);
+  });
```

Comment:
basically, this code toggles the class name on the popover arrow elements, preserving the original classes... there is no way to do this with popover API